In [1]:
import pandas as pd

# Explore Occupations likely to be automated

## About the Data

The following dataset I found via Google Data Search. I chose this data set becuase it is relatively simple, yet I find it may add value to my future research. 

Below is some information on the source: 

> Dataset updated Feb 17, 2020
Dataset provided by
data.world

>Authors
William Nedds

>Description ([Link to the DataSet Source](https://datasetsearch.research.google.com/search?query=download%20us%20IT%20job%20stats%20tabular&docid=eUPlmll93JawI7h9AAAAAA%3D%3D&filters=WyJbXCJmaWxlX2Zvcm1hdF9jbGFzc1wiLFtcIjFcIl1dIl0%3D&property=ZmlsZV9mb3JtYXRfY2xhc3M%3D))

>About this project
As part of an ongoing project on automation in the workforce, I have combined the automation probability data in my previous dataset with a breakdown of the number of jobs in each occupation by state. Automation Probability is from [THE FUTURE OF EMPLOYMENT: HOW
SUSCEPTIBLE ARE JOBS TO
COMPUTERISATION?∗ Carl Benedikt Frey
and Michael A. Osborne](https://www.oxfordmartin.ox.ac.uk/downloads/academic/The_Future_of_Employment.pdf)




> Carl Benedikt Frey†
and Michael A. Osborne‡
September 17, 2013
.
Abstract
We examine how susceptible jobs are to computerisation. To assess this, we begin by implementing a novel methodology to estimate
the probability of computerisation for 702 detailed occupations, using a
Gaussian process classifier. Based on these estimates, we examine expected impacts of future computerisation on US labour market outcomes,
with the primary objective of analysing the number of jobs at risk and
the relationship between an occupation’s probability of computerisation,
wages and educational attainment. According to our estimates, about 47
percent of total US employment is at risk. We further provide evidence
that wages and educational attainment exhibit a strong negative relationship with an occupation’s probability of computerisation.

## Quick preview of the data

In [11]:
occupations = pd.read_csv("data/raw_state_automation_data.csv", encoding = "ISO-8859-1")

.shape
(rows, cols)

In [12]:
occupations.shape

(702, 54)

**Indicates 702 rows and 54 Columns**

In [13]:
occupations.head()

SOC                           Occupation  Probability  Alabama  Alaska  \
0  11-1011                     Chief Executives        0.015     1030     760   
1  11-1021      General and Operations Managers        0.160    26930    6490   
2  11-2011  Advertising and Promotions Managers        0.039       50      40   
3  11-2021                   Marketing Managers        0.014      530     200   
4  11-2022                       Sales Managers        0.013     2510     400   

   Arizona  Arkansas  California  Colorado  Connecticut  ...  South Dakota  \
0     5750      2710       31150       880         1410  ...           560   
1    43300     20680      261780     41540        33280  ...          3730   
2      470       110        3760       480          300  ...             0   
3     4790      1090       33390      3060         4970  ...            60   
4    10650      2650       69180      4570         7040  ...           300   

   Tennessee   Texas   Utah  Vermont  Virginia  Washington  West Virginia  \
0       5460    5890   3650      280      6320        5910            980   
1      44400  168610  36200     2760     52380       43760          10200   
2        670    1210    380       40       240         640             40   
3       3400    9570   2320      380      3840        5830            260   
4       8890   22310   3360      480      5360        7390            600   

   Wisconsin  Wyoming  
0       3740      160  
1      32350     4840  
2        200        0  
3       2980       30  
4       5730      180  

[5 rows x 54 columns]

### Basic views of data

**Listing all columns**

In [26]:
occupations.iloc[0]

SOC                              11-1011
Occupation              Chief Executives
Probability                        0.015
Alabama                             1030
Alaska                               760
Arizona                             5750
Arkansas                            2710
California                         31150
Colorado                             880
Connecticut                         1410
Delaware                             340
District of Columbia                2840
Florida                            14120
Georgia                             6750
Hawaii                              1840
Idaho                               1400
Illinois                           17440
Indiana                             4950
Iowa                                2050
Kansas                              4410
Kentucky                            3440
Louisiana                           1010
Maine                                920
Maryland                            1800
Massachusetts   

**Listing the 2nd (index 1) column**

In [28]:
occupations.iloc[:, 1]

0                               Chief Executives
1                General and Operations Managers
2            Advertising and Promotions Managers
3                             Marketing Managers
4                                 Sales Managers
                         ...                    
697      Pump Operators; Except Wellhead Pumpers
698                             Wellhead Pumpers
699    Refuse and Recyclable Material Collectors
700                   Mine Shuttle Car Operators
701            Tank Car; Truck; and Ship Loaders
Name: Occupation, Length: 702, dtype: object

In [31]:
occupations.iloc[:13,1]

0                                      Chief Executives
1                       General and Operations Managers
2                   Advertising and Promotions Managers
3                                    Marketing Managers
4                                        Sales Managers
5             Public Relations and Fundraising Managers
6                      Administrative Services Managers
7             Computer and Information Systems Managers
8                                    Financial Managers
9                        Industrial Production Managers
10                                  Purchasing Managers
11    Transportation; Storage; and Distribution Mana...
12                   Compensation and Benefits Managers
Name: Occupation, dtype: object

### Exploring Ohio Data

In [32]:
occupations.head()

SOC                           Occupation  Probability  Alabama  Alaska  \
0  11-1011                     Chief Executives        0.015     1030     760   
1  11-1021      General and Operations Managers        0.160    26930    6490   
2  11-2011  Advertising and Promotions Managers        0.039       50      40   
3  11-2021                   Marketing Managers        0.014      530     200   
4  11-2022                       Sales Managers        0.013     2510     400   

   Arizona  Arkansas  California  Colorado  Connecticut  ...  South Dakota  \
0     5750      2710       31150       880         1410  ...           560   
1    43300     20680      261780     41540        33280  ...          3730   
2      470       110        3760       480          300  ...             0   
3     4790      1090       33390      3060         4970  ...            60   
4    10650      2650       69180      4570         7040  ...           300   

   Tennessee   Texas   Utah  Vermont  Virginia  Washington  West Virginia  \
0       5460    5890   3650      280      6320        5910            980   
1      44400  168610  36200     2760     52380       43760          10200   
2        670    1210    380       40       240         640             40   
3       3400    9570   2320      380      3840        5830            260   
4       8890   22310   3360      480      5360        7390            600   

   Wisconsin  Wyoming  
0       3740      160  
1      32350     4840  
2        200        0  
3       2980       30  
4       5730      180  

[5 rows x 54 columns]

In [37]:
occupations.loc[[0,1,3],'Ohio']

0     5340
1    64710
3     5350
Name: Ohio, dtype: int64

In [65]:
occupations.Ohio.describe()

count       702.000000
mean       6761.111111
std       15816.796789
min           0.000000
25%         360.000000
50%        1585.000000
75%        6050.000000
max      163790.000000
Name: Ohio, dtype: float64

### Let's get the columns I care about:


In [38]:
occupations.loc[:, ['Occupation', 'Probability', 'Ohio']]

Occupation  Probability   Ohio
0                             Chief Executives        0.015   5340
1              General and Operations Managers        0.160  64710
2          Advertising and Promotions Managers        0.039    460
3                           Marketing Managers        0.014   5350
4                               Sales Managers        0.013  11410
..                                         ...          ...    ...
697    Pump Operators; Except Wellhead Pumpers        0.900    210
698                           Wellhead Pumpers        0.840    580
699  Refuse and Recyclable Material Collectors        0.930   6190
700                 Mine Shuttle Car Operators        0.370      0
701          Tank Car; Truck; and Ship Loaders        0.720    260

[702 rows x 3 columns]

### Creating a new DataFrame based on key columns

In [50]:
ohio_occupations = occupations.loc[:, ['Occupation', 'Probability', 'Ohio']]

In [110]:
ohio_occupations.columns = ['Title','Probability', 'Job_Count' ]

In [111]:
 ohio_occupations.head()

Title  Probability  Job_Count
0                     Chief Executives        0.015       5340
1      General and Operations Managers        0.160      64710
2  Advertising and Promotions Managers        0.039        460
3                   Marketing Managers        0.014       5350
4                       Sales Managers        0.013      11410

#### Some questions I have: 
* which occupations have the highest probability of being automated?
* which occupations have the lowest ratio, between highest prability and highest number of jobs in Ohio? 
    * For example (Probability = High, No of Jobs = High)

In [112]:
ohio_occupations.Job_Count.median()

1585.0

In [113]:
ohio_occupations.Job_Count.max()

163790

In [114]:
# showing the top 5 rows
ohio_occupations.head()

Title  Probability  Job_Count
0                     Chief Executives        0.015       5340
1      General and Operations Managers        0.160      64710
2  Advertising and Promotions Managers        0.039        460
3                   Marketing Managers        0.014       5350
4                       Sales Managers        0.013      11410

In [115]:
# showing occupations with over 10,000 jobs
ohio_occupations.loc[ohio_occupations.Job_Count > 100000]

Title  Probability  Job_Count
317  Combined Food Preparation and Serving Workers;...         0.92     163790
364                                           Cashiers         0.97     117390
368                                Retail Salespersons         0.92     157460
693  Laborers and Freight; Stock; and Material Move...         0.85     111230

### Interpretation of Ohio Job Count Column
* There are 702 different occupations, some with 0 job count
* 75% of the Occuaptions have a job count of 6,050, which also drives the mean value


In [127]:
ohio_occupations.Job_Count.describe()


count       702.000000
mean       6761.111111
std       15816.796789
min           0.000000
25%         360.000000
50%        1585.000000
75%        6050.000000
max      163790.000000
Name: Job_Count, dtype: float64

### Ohio Automatable Occupations Report I

| Prob. of Automation | No. of Jobs per Occupation |
|:----------------------------------:|:------------------------:|
|                > 60%               |    >30,000    |

In [117]:
# showing occupations with over 10,000 jobs each, 
# and a probability greater than 80% of being automated
ohio_occupations.loc[(ohio_occupations.Job_Count >30000) & 
                     (ohio_occupations.Probability > .60)].sort_values(by='Probability', ascending=False)

Title  Probability  Job_Count
417            Shipping; Receiving; and Traffic Clerks         0.98      34030
387       Bookkeeping; Accounting; and Auditing Clerks         0.98      61590
634  Inspectors; Testers; Sorters; Samplers; and We...         0.98      31040
559                                    Team Assemblers         0.97      41740
364                                           Cashiers         0.97     117390
313                                  Cooks; Restaurant         0.96      37660
430                             Office Clerks; General         0.96      92350
423  Secretaries and Administrative Assistants; Exc...         0.96      76950
329             Landscaping and Groundskeeping Workers         0.95      35550
48                            Accountants and Auditors         0.94      42890
319                             Waiters and Waitresses         0.94      98150
368                                Retail Salespersons         0.92     157460
317  Combined Food Preparation and Serving Workers;...         0.92     163790
456                              Construction Laborers         0.88      35090
374  Sales Representatives; Wholesale and Manufactu...         0.85      48290
693  Laborers and Freight; Stock; and Material Move...         0.85     111230
305                                    Security Guards         0.84      32280
422                                Medical Secretaries         0.81      39610
666            Heavy and Tractor-Trailer Truck Drivers         0.79      70740
667           Light Truck or Delivery Services Drivers         0.69      35060
326  Janitors and Cleaners; Except Maids and Housek...         0.66      85470
418                     Stock Clerks and Order Fillers         0.64      84580
541            Maintenance and Repair Workers; General         0.64      53720
309  First-Line Supervisors of Food Preparation and...         0.63      34680

### Ohio Automatable Occupations Report II

| Prob. of Automation | No. of Jobs per Occupation |
|:----------------------------------:|:------------------------:|
|                > 97%               |    >2,000 and <70,000    |

In [141]:
# showing occupations with over 10,000 jobs each, 
# and a probability greater than 80% of being automated
ohio_occupations.loc[(ohio_occupations.Job_Count <70000 ) & 
                     ( ohio_occupations.Job_Count >2000 ) &
                     (ohio_occupations.Probability > .97)].sort_values(by='Probability', ascending=False)

Title  Probability  Job_Count
408                           Cargo and Freight Agents         0.99       2690
54                              Insurance Underwriters         0.99       4860
59                                       Tax Preparers         0.99       2040
172        Title Examiners; Abstractors; and Searchers         0.99       2130
190                                Library Technicians         0.99       4000
380                                      Telemarketers         0.99      13510
425                                  Data Entry Keyers         0.99       5920
403                                New Accounts Clerks         0.99       2290
35      Claims Adjusters; Examiners; and Investigators         0.98       9840
639  Packaging and Filling Machine Operators and Te...         0.98      17460
634  Inspectors; Testers; Sorters; Samplers; and We...         0.98      31040
428      Insurance Claims and Policy Processing Clerks         0.98      17600
421                                  Legal Secretaries         0.98       7330
417            Shipping; Receiving; and Traffic Clerks         0.98      34030
391                                            Tellers         0.98      19260
404                                       Order Clerks         0.98      10050
392                                   Brokerage Clerks         0.98       2910
51                                     Credit Analysts         0.98       2810
390                                 Procurement Clerks         0.98       3840
387       Bookkeeping; Accounting; and Auditing Clerks         0.98      61590
367                                 Parts Salespersons         0.98      10500
57                                       Loan Officers         0.98      11980
665                               Driver/Sales Workers         0.98      22220